enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,256 @@
+  - "@eranco"
+  - "@romfreiman"
+  - "@tsorya"
+reviewers:
+  - TBD
+  - "@mrunalp"
+  - "@markmc"
+  - "@deads2k"
+  - "@wking"
+  - "@eparis"
+  - "@hexfusion"
+approvers:
+  - TBD
+creation-date: 2020-12-13
+last-updated: 2020-12-13
+status: implementable
+see-also:
+  - https://github.com/openshift/enhancements/pull/560
+  - https://github.com/openshift/enhancements/pull/302
+---
+
+# Single Node Production Edge Installer - bootstrap in place
+
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+As we add the new [`single-node production deployment`](https://github.com/openshift/enhancements/pull/560/files)
+we need a way to insall such cluster without extra node dependency for bootstrap.
+
+This enhancement describes the flow for installing Single Node Openshift using liveCD that perform the bootstrap logic and later reboot to become the single node.
+
+## Motivation
+
+Currently, all OpenShift installations use an auxiliary bootstrap node.
+The bootstrap node creates a temporary control plane that is required for launching the actual cluster.
+
+Removing the bootstrap node is a key for the 5G RAN use case and to the more general topic of single-node OCP clusters for edge use cases.
+
+Requiring the axillary node for installing a Single Node Openshift brings to play some requirements:
+1. The obvious additional node, in most cases there isn't additional Hardware/VM at the edge site.
+2. Requires external dependencies:
+   a. Load balancer (only for bootstrap phase)
+   b. Requires DNS (configured per installation)
+3. Cannot use Bare Metal IPI:
+   a. Adds irrelevant dependencies - vips, keepalived, mdns
+   b. Requires same L2 between bootstrap and the Single Node Openshift
+
+### Goals
+
+* This enhancement describes an approach for insallting Single Node OpenShift 
+  for production use.
+* Minimal changes to openshift installer, this implementation shouldn't effect existing deployment flows.
+* Installation should result a clean Single Node Openshift, without bootstrap leftovers.
+* Rely on a livecd, so it would be able to trigger the full installation using virtual media.
+* Self-Contained bootstrap, The bootstrap flow is expressed as a single Ignition config which is layered on top of a pristine RHCOS image. It's important that we limit our use of this pattern to just the Ignition configs - as opposed to including other assets directly in the ISO image. 
+
+### Non-Goals
+
+* This enhancement does not address similar insallation flow for 3 nodes cluster.
+* This enhancement does not address high-availability for single-node
+  deployments.
+* This enhancement does not address single-node-developer cluster-profile installation.
```

Comment:
This enhancement does not address cloud deployment for bootstrap in place (since livecd cannot be used). It will be addressed as part of future enhancements

Reply:
Another non-goal is:

> * This enhancement does not address updating single-node clusters.

We'd need a follow-up enhancement to decide if that was possible, and it's fairly orthogonal to the installation case.

Reply:
Does using cloud instances really preclude the use of a boot image that's equivalent to a liveCD?

Reply:
It is not possible to attach a CD-ROM and run your ISO in EC2.
There are possible workarounds such as mounting some directories as tmpfs during the bootstrap phase as @cgwalters suggested.

Reply:
Added "Allow bootstrap-in-place in cloud environment" to alternatives